sql/core/src/test/resources/sql-tests/inputs/pgSQL/numeric.sql

```diff
@@ -0,0 +1,1098 @@
+-- select div (70.0,70) ;
+select 70.0 / 70 ;
+select 12345678901234567890 % 123;
+-- [SPARK-2659] HiveQL: Division operator should always perform fractional division
+-- select 12345678901234567890 DIV 123;
+-- select div(12345678901234567890, 123);
+-- select div(12345678901234567890, 123) * 123 + 12345678901234567890 % 123;
+
+-- [SPARK-28007] Caret operator (^) means bitwise XOR in Spark and exponentiation in Postgres
+--
+-- Test code path for raising to integer powers
+--
+
+-- select 10.0 ^ -2147483648 as rounds_to_zero;
+-- select 10.0 ^ -2147483647 as rounds_to_zero;
+-- select 10.0 ^ 2147483647 as overflows;
+-- select 117743296169.0 ^ 1000000000 as overflows;
+
+-- cases that used to return inaccurate results
+-- select 3.789 ^ 21;
+-- select 3.789 ^ 35;
+-- select 1.2 ^ 345;
+-- select 0.12 ^ (-20);
+
+-- cases that used to error out
+-- select 0.12 ^ (-25);
+-- select 0.5678 ^ (-85);
+
+--
+-- Tests for raising to non-integer powers
+--
+
+-- special cases
+-- select 0.0 ^ 0.0;
+-- select (-12.34) ^ 0.0;
+-- select 12.34 ^ 0.0;
+-- select 0.0 ^ 12.34;
+
+-- NaNs
+-- select 'NaN'::numeric ^ 'NaN'::numeric;
+-- select 'NaN'::numeric ^ 0;
+-- select 'NaN'::numeric ^ 1;
+-- select 0 ^ 'NaN'::numeric;
+-- select 1 ^ 'NaN'::numeric;
+
+-- invalid inputs
+-- select 0.0 ^ (-12.34);
+-- select (-12.34) ^ 1.2;
+
+-- cases that used to generate inaccurate results
+-- select 32.1 ^ 9.8;
+-- select 32.1 ^ (-9.8);
+-- select 12.3 ^ 45.6;
+-- select 12.3 ^ (-45.6);
+
+-- big test
+-- select 1.234 ^ 5678;
+
+--
+-- Tests for EXP()
+--
+
+-- special cases
+select exp(0.0);
+select exp(1.0);
+-- [SPARK-28316] EXP returns double type for decimal input
+-- [SPARK-28318] Decimal can only support precision up to 38
+-- select exp(1.0::numeric(71,70));
+
+-- cases that used to generate inaccurate results
+select exp(32.999);
+select exp(-32.999);
+select exp(123.456);
+select exp(-123.456);
+
+-- big test
+select exp(1234.5678);
+
+--
+-- Tests for generate_series
+--
+select * from range(cast(0.0 as decimal(38, 18)), cast(4.0 as decimal(38, 18)));
+select * from range(cast(0.1 as decimal(38, 18)), cast(4.0 as decimal(38, 18)), cast(1.3 as decimal(38, 18)));
+select * from range(cast(4.0 as decimal(38, 18)), cast(-1.5 as decimal(38, 18)), cast(-2.2 as decimal(38, 18)));
+-- Trigger errors
+-- select * from generate_series(-100::numeric, 100::numeric, 0::numeric);
+-- select * from generate_series(-100::numeric, 100::numeric, 'nan'::numeric);
+-- select * from generate_series('nan'::numeric, 100::numeric, 10::numeric);
+-- select * from generate_series(0::numeric, 'nan'::numeric, 10::numeric);
+-- [SPARK-28007] Caret operator (^) means bitwise XOR in Spark and exponentiation in Postgres
+-- Checks maximum, output is truncated
+-- select (i / (10::numeric ^ 131071))::numeric(1,0)
+-- 	from generate_series(6 * (10::numeric ^ 131071),
+-- 			     9 * (10::numeric ^ 131071),
+-- 			     10::numeric ^ 131071) as a(i);
+-- Check usage with variables
```

Comment:
I'd like to skip these test.

Reply:
because `generate_series` unsupported?

Reply:
See: https://issues.apache.org/jira/browse/SPARK-27767

Reply:
ur, I see. Thanks!